vertx-sql-client/src/main/java/io/vertx/sqlclient/Tuple.java

```diff
@@ -428,7 +428,16 @@ default JsonObject getJsonObject(int pos) {
    * @return the value
    */
   default JsonArray getJsonArray(int pos) {
-    return (JsonArray) getValue(pos);
+    Object val = getValue(pos);
+    if (val == null) return null;
+    if (val instanceof JsonArray) return (JsonArray) val;
+    if (val instanceof Collection) {
+      return new JsonArray(new ArrayList<>(((Collection<?>) val)));
+    }
+    if (val.getClass().isArray()) {
+      return IntStream.range(0, Array.getLength(val)).mapToObj(i -> Array.get(val, i)).collect(JsonArray::new, JsonArray::add, (o1, o2) -> {});
```

Comment:
It would be better to avoid the stream API on this part. It is not really useful and allocates much more memory.

Reply:
This applies to the other,  similar changes